### PR TITLE
fix(response): reoder exception handling to correctly parse a broken response

### PR DESCRIPTION
**Pull request by SlowMo24**

### CODE REVIEW

File: ohsome/test/test_exceptions.py [hunk 1]
```diff
@@ -43,6 +43,34 @@ def test_timeout_error(base_client):
     )
 
 
+def test_borken_response_timeout_error(base_client):
```
Comment (matthiasschaub) [inline ohsome/test/test_exceptions.py:46]:
```suggestion
def test_broken_response_timeout_error(base_client):
```

File: ohsome/test/test_exceptions.py [hunk 2]
```diff
@@ -43,6 +43,34 @@ def test_timeout_error(base_client):
...
+    """
+    Test whether an OhsomeException is raised, if the ohsome API contains a JSONDecodeError
+    :return:
+    """
```
Comment (matthiasschaub) [inline ohsome/test/test_exceptions.py:50]:
```suggestion
    """Test whether an OhsomeException is raised in case of a JSONDecodeError."""
```

**Review by matthiasschaub — COMMENTED** [review]
I left two minor suggestions.

In general, it would be nice to do only one thing in the try-block.

Then it is easier to parse the exceptions. What can go wrong where.

**Comment by SlowMo24** [comment]
> In general, it would be nice to do only one thing in the try-block.
> 
> Then it is easier to parse the exceptions. What can go wrong where.

You suggest placing line 341 https://github.com/GIScience/ohsome-py/pull/164/files#diff-0684a040fd4741e6267194be04e688f118c72fdae4256902f407e9a6c11dac2aL341 into a separate try block?

**Comment by matthiasschaub** [comment]
> > In general, it would be nice to do only one thing in the try-block.
> > Then it is easier to parse the exceptions. What can go wrong where.
> 
> You suggest placing line 341 https://github.com/GIScience/ohsome-py/pull/164/files#diff-0684a040fd4741e6267194be04e688f118c72fdae4256902f407e9a6c11dac2aL341 into a separate try block?

Yes, or even:
```python
response = self._session().post(url=self._url, data=self._parameters)
try:
    response.raise_for_status(
except:
    ...
try:
    response.json()
except:
    ...
```

So that one knows that exceptions caught in one block belong only to the http response or json parsing and not to `_session().post`.

**Comment by SlowMo24** [comment]
> > > In general, it would be nice to do only one thing in the try-block.
> > > Then it is easier to parse the exceptions. What can go wrong where.
> > 
> > 
> > You suggest placing line 341 https://github.com/GIScience/ohsome-py/pull/164/files#diff-0684a040fd4741e6267194be04e688f118c72fdae4256902f407e9a6c11dac2aL341 into a separate try block?
> 
> Yes, or even:
> 
> ```python
> response = self._session().post(url=self._url, data=self._parameters)
> try:
>     response.raise_for_status(
> except:
>     ...
> try:
>     response.json()
> except:
>     ...
> ```
> 
> So that one knows that exceptions caught in one block belong only to the http response or json parsing and not to `_session().post`.

thanks, I had the same gut feeling during coding but thought it was too complicated. I have made this change now, please re-review

**Pull request merged**